### PR TITLE
Bind to qpid queue to specific events for Foreman

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ehelms** [comment]
@akofink I made a few changes here. The testing for this is not straight forward yet as it's a bit manual. Assuming you have minishift running and a deployment already

```
eval $(minishift docker-env)
ansible-playbook build.yml -e image=foreman
docker tag foreman $(minishift openshift registry)/foreman/foreman:latest
docker push $(minishift openshift registry)/foreman/foreman:latest
```

Go into the web UI, edit the deployment for Foreman and change the image reference to:

```
172.30.1.1:5000/foreman/foreman:latest
```

Re-deploy Foreman, check the logs and check the ListenOnCandlepinEvents

File: containers/images/foreman/Dockerfile [hunk 1]
```diff
@@ -32,9 +32,8 @@ COPY container-assets/database.yml /etc/foreman/database.yml
 COPY container-assets/entrypoint.sh /usr/bin/entrypoint.sh
 COPY container-assets/start_foreman.sh /usr/bin/start_foreman.sh
 COPY container-assets/wait_on_postgres.py /usr/bin/wait_on_postgres.py
-COPY container-assets/create_qpid_queue.sh /usr/bin/create_qpid_queue.sh
 
-USER foreman
```
Comment (ekohl) [inline containers/images/foreman/Dockerfile]: Is this change intentional?

Comment (ehelms) [reply containers/images/foreman/Dockerfile]: Yes. It is causing more problems than solving and is not required to run the container. This will be nice to have when we want to push all containers to not have to run as root.

File: containers/images/foreman/container-assets/create_qpid_queue.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+---
+- hosts: localhost
+  gather_facts: no
+  tasks:
+    - set_fact:
+        qpid_port: "{{ lookup('env', 'QPID_SERVICE_PORT') }}"
+
+    - set_fact:
```
Comment (ekohl) [inline containers/images/foreman/container-assets/create_qpid_queue.yml]: You can combine these 2 `set_fact` into a single block.

File: containers/images/foreman/container-assets/create_qpid_queue.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+---
+- hosts: localhost
+  gather_facts: no
+  tasks:
+    - set_fact:
+        qpid_port: "{{ lookup('env', 'QPID_SERVICE_PORT') }}"
```
Comment (ekohl) [inline containers/images/foreman/container-assets/create_qpid_queue.yml]: Should you pipe it through `int` here instead of everywhere else?

File: containers/images/foreman/container-assets/create_qpid_queue.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      wait_for:
+        port: "{{ qpid_port | int }}"
+        host: "{{ qpid_host }}"
+        timeout: 600
+
+    - name: "Check for qpid queue"
+      command: "qpid-config -b {{ qpid_host }}:{{ qpid_port }} queues katello_event_queue"
+      register: queue_exist
+
+    - name: "Add katello_event_que"
```
Comment (ekohl) [inline containers/images/foreman/container-assets/create_qpid_queue.yml]: s/que/queue/

File: containers/images/foreman/container-assets/create_qpid_queue.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+      command: "qpid-config -b {{ qpid_host }}:{{ qpid_port }} add queue katello_event_queue --durable"
+      when: '"katello_event_queue" in queue_exist'
```
Comment (ekohl) [inline containers/images/foreman/container-assets/create_qpid_queue.yml]: Should this be `not in`?

**Comment by ehelms** [comment]
Updated

**Comment by ehelms** [comment]
[test containers]

**Comment by ehelms** [comment]
[test containers]

File: containers/images/foreman/container-assets/create_qpid_queue.yml [hunk 6]
```diff
@@ -0,0 +1,33 @@
+---
+- hosts: localhost
+  gather_facts: no
+  tasks:
+    - set_fact:
+        qpid_port: "{{ lookup('env', 'QPID_SERVICE_PORT') | int }}"
+        qpid_host: "{{ lookup('env', 'QPID_SERVICE_HOST') }}"
+
+    - debug:
+        msg: "{{ qpid_host }}:{{ qpid_port | int }}"
```
Comment (ekohl) [inline containers/images/foreman/container-assets/create_qpid_queue.yml]: The `| int` is redundant
